matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/auth/login/DefaultLoginWizard.kt

```diff
@@ -136,4 +136,14 @@ internal class DefaultLoginWizard(
         // Set to null?
         // resetPasswordData = null
     }
+
+    private fun PendingSessionData.readResetPasswordDataOrThrow(newPassword: String?): Pair<AddThreePidRegistrationResponse, String> {
+        return when (resetPasswordData) {
+            null -> throw IllegalStateException("developer error, no reset password in progress")
+            else -> {
+                val password = newPassword ?: resetPasswordData.newPassword ?: throw IllegalStateException("developer error, no new password set")
```

Comment:
Do we want to enforce that `newPassword` and `resetPasswordData.newPassword` are not not null simultaneously? Or are the same?